tests/unit/lms/services/launch_verifier_test.py

```diff
@@ -169,48 +173,71 @@ def launch_verifier(self, pyramid_request):
-        pyramid_request.path_url += "/"
-
-        # Add the OAuth 1 params (version, nonce, timestamp, ...)
-        oauthlib_client = oauthlib.oauth1.Client(
-            pyramid_request.params["oauth_consumer_key"]
-        )
-        oauthlib_request = oauthlib.common.Request(
-            pyramid_request.url, pyramid_request.method
-        )
-        pyramid_request.params = dict(
-            oauthlib_client.get_oauth_params(oauthlib_request)
-        )
+        def set_url_from_params():
+            parsed_url = urlparse(pyramid_request.url)
+            parsed_url = parsed_url._replace(query=urlencode(pyramid_request.params))
+            pyramid_request.url = parsed_url.geturl()
 
-        sign(pyramid_request)
+        pyramid_request.set_url_from_params = set_url_from_params
 
         return pyramid_request
 
+    @pytest.fixture(autouse=True)
+    def oauth1(self, pyramid_request):
+        oauth1 = OAuth1Assistant(pyramid_request)
+        oauth1.add_oauth_params()  # Add the OAuth 1 params (version, nonce, timestamp, ...)
+        oauth1.sign()  # Sign the request
+
+        return oauth1
+
     @pytest.fixture
-    def pylti(self, patch):
-        pylti = patch("lms.services.launch_verifier.pylti")
-        pylti.common.LTIException = LTIException
-        return pylti
+    def oauth1_endpoint(self, launch_verifier):
+        oauth1_endpoint = create_autospec(
+            SignatureOnlyEndpoint, instance=True, spec_set=True
+        )
+        oauth1_endpoint.validate_request.return_value = True, sentinel.request
 
+        # pylint: disable=protected-access
+        launch_verifier._oauth1_endpoint = oauth1_endpoint
 
-def sign(pyramid_request):
-    """Add or replace pyramid_request's OAuth 1 signature param."""
-    oauthlib_client = oauthlib.oauth1.Client(
-        pyramid_request.params["oauth_consumer_key"], "TEST_SECRET"
-    )
-    oauthlib_request = oauthlib.common.Request(
-        pyramid_request.url, pyramid_request.method, body=pyramid_request.params
-    )
-    pyramid_request.params["oauth_signature"] = oauthlib_client.get_oauth_signature(
-        oauthlib_request
-    )
+        return oauth1_endpoint
+
+
+class OAuth1Assistant:
+    def __init__(self, pyramid_request):
+        self.client = oauthlib.oauth1.Client(
+            client_key=pyramid_request.params["oauth_consumer_key"],
+            client_secret="TEST_SECRET",
+        )
+        self.pyramid_request = pyramid_request
+
+    def _get_request(self):
+        parsed_url = urlparse(self.pyramid_request.url)
+        parsed_url = parsed_url._replace(query=None)
+
+        return oauthlib.common.Request(
+            # This all seems very wrong, for actual requests we expect the URL to
+            # contain the query string, here we remove it
+            parsed_url.geturl(),
+            self.pyramid_request.method,
+            # And set the query string as the body... in any case it works
+            body=self.pyramid_request.params,
+        )
```

Comment:
We're keeping the old weird way of doing the signing but we will need to put the results back on the URL later...